### ISSUE TITLE
Closed-source options

### ISSUE THREAD

**Issue by richardthombs** [rafinskipg/git-changelog#34]
Sorry to be the evil-one, but I would like to use this tool to generate change logs for a closed-source project. To do this, I need
- [ ] An option to turn off the github commit links, as our readers won't have access to the repo.
- [ ] An option to override the salute message.

I'm happy to write the pull request for this, but I would like some advice on how to pass options object through to the relevant parts of the code, as I'm new to the `require` style of programming used in this project.

**Comment by rafinskipg** [comment]
Hi, @richardthombs , yeah you can write a pull request if you want. 

All the default options are specified in `/tasks/default.js` .

In the `/tasks/command.js` you can see an example of how this defaults are overriden:

```javascript
var defaults = require('./defaults');
var _ = require('lodash');
var options = _.defaults({}, defaults);
```
The `lodash` `_.defaults` methods, clones the object and assigns some extra overriden values. (See lodash.com docs)

After having the `options` object, we assign some new values from the CLI with `options.whatever = whatever`. Then we send them to the generator:

```javascript
git_changelog.generate(options)
```

The generate method is in `/tasks/lib/generate`. This first of all invokes the `.init` method, that configures the library with the options in the ` this.initOptions(params);` method. (That is at `/tasks/lib/init-options.js`).

I see a bit of duplicity here `this.options = _.defaults(params, defaults);`. But whatever, here are finally setted the options in the library.

Your new options should arrive to this point. Then use them as needed accesing to `this.options.myNewOption` at any point of the application. As all is a big object containing all the methods.

Sorry for the delay.

Be sure to add tests. :+1:  @JohnnyEstilles  have worked hard on getting a good coverage

**Comment by richardthombs** [comment]
Thanks for the info, I'll take a crack at it soon.

**Comment by rafinskipg** [comment]
@richardthombs  With the new template system you can add your custom salute message and your custom link formatting. :+1:

**Comment by richardthombs** [comment]
What else could I ask for on Christmas day! THANK YOU :)

**Comment by rafinskipg** [comment]
Merged and 1.2.0 released! Thank you too! merry xmas
